lib/ansible/plugins/action/copy.py

```diff
@@ -155,6 +155,12 @@ def run(self, tmp=None, task_vars=None):
             # Generate a hash of the local file.
             local_checksum = checksum(source_full)
 
+            # Get the local mode and set if not user defined
+            # https://github.com/ansible/ansible-modules-core/issues/1124
+            if not self._task.args.get('mode', None):
+                lmode = oct(os.stat(source_full).st_mode & 0o777)
```

Comment:
Instead of oct, use '0%03o' % os.stat(source_full).st_mode & 0o777 for python3 compat.

I just remembered there's also a function in the python stat module which may do the & 0o777 a bit prettier:

```
import stat
lmode = '0%03o' % stat.S_IMODE(os.stat(source_full).st_mode)
```